contract/src/lib.rs

```diff
@@ -49,15 +48,69 @@ pub enum ProtocolContractState {
 }
 
 #[near_bindgen]
-#[derive(BorshDeserialize, BorshSerialize, PanicOnDefault)]
+#[derive(BorshDeserialize, BorshSerialize, Debug)]
+pub enum VersionedMpcContract {
+    V0(MpcContract),
+}
+
+impl Default for VersionedMpcContract {
+    fn default() -> Self {
+        env::panic_str("Calling default not allowed.");
+    }
+}
+
+#[derive(BorshDeserialize, BorshSerialize, PanicOnDefault, Debug)]
 pub struct MpcContract {
     protocol_state: ProtocolContractState,
     pending_requests: LookupMap<[u8; 32], Option<(String, String)>>,
     request_counter: u32,
 }
 
-#[near_bindgen]
 impl MpcContract {
+    fn add_request(&mut self, payload: &[u8; 32], signature: &Option<(String, String)>) {
+        if self.request_counter > 8 {
+            env::panic_str("Too many pending requests. Please, try again later.");
+        }
+        if !self.pending_requests.contains_key(payload) {
+            self.request_counter += 1;
+        }
+        self.pending_requests.insert(payload, signature);
+    }
+
+    fn remove_request(&mut self, payload: &[u8; 32]) {
+        self.pending_requests.remove(payload);
+        self.request_counter -= 1;
+    }
+
+    fn add_signature(&mut self, payload: &[u8; 32], signature: (String, String)) {
+        if self.pending_requests.contains_key(payload) {
+            self.pending_requests.insert(payload, &Some(signature));
+        }
+    }
+
+    fn clean_payloads(&mut self, payloads: Vec<[u8; 32]>, counter: u32) {
+        log!("clean_payloads");
+        for payload in payloads.iter() {
+            self.pending_requests.remove(payload);
+        }
+        self.request_counter = counter;
+    }
+
+    pub fn test_init() -> Self {
+        MpcContract {
+            protocol_state: ProtocolContractState::Initializing(InitializingContractState {
+                candidates: Candidates::default(),
+                threshold: 2,
+                pk_votes: PkVotes::new(),
+            }),
+            pending_requests: LookupMap::new(b"m"),
+            request_counter: 2,
+        }
+    }
```

Comment:
This should not be apart of the contract itself since that will bloat the code size and increase the storage staking cost of deploying this contract